addons/mrp/models/mrp_bom.py

```diff
@@ -446,8 +446,16 @@ def _get_product_catalog_order_data(self, products, **kwargs):
         product_catalog = super()._get_product_catalog_order_data(products, **kwargs)
         for product in products:
             product_catalog[product.id] |= self._get_product_price_and_data(product)
+            product_catalog[product.id] |= self._get_product_quantity(product)
         return product_catalog
 
+    def _get_product_quantity(self, product):
+        self.ensure_one()
+        line = self.bom_line_ids.filtered(lambda x: x.product_id.id == product.id)
+        if not line:
+            return {}
+        return {'quantity': line.product_qty}
+
```

Comment:
what is this for?

Reply:
for this: 
![image](https://github.com/odoo/odoo/assets/39632709/fee23458-6984-43f3-8d11-b4667867415c)
At the BoM and MO, when we call the `In the BoM` or `In the MO`, the quantity is not passed, so, in this scenario, the products are not marked with "selected" and there are no quantities for each product.

Reply:
This sounds like a bug that should be fixed in saas-17.2 (it appears to be broken in that version too) :thinking: MrpBomLine already has `_get_product_catalog_lines_data()` set up to calculate + return the quantity so something must be wrong within it. Same bug exists with the by-product's catalog